app/controllers/users_controller.rb

```diff
@@ -289,4 +285,36 @@ def registration_successful
     end
   end
 
+  def edit_by_project_admin
+    @user = User.find(params[:id])
+    authorize @user
+    @projects = Admin::Project.all_sorted
+  end
+  def update_by_project_admin
+    if params[:commit] == "Cancel"
+      # FIXME: ugly hack
+      # if the Cancel request came from a form generated by
+      # the preferences action then redirect to /home
+      if request.env["HTTP_REFERER"] =~ /preferences/
+        redirect_to :home
+      else
+        redirect_to users_path
+      end
```

Comment:
In this case, the update_by_project_admin should never come from a preferences page. So I think you can remove the check of the referrer and just redirect to the index page.